src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,33 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.ValidationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_LOGIN_AND_PASSWORD_LENGTH = 6;
+    private static final int MIN_USER_AGE = 18;
+    private static final int MIN_ID_LENGTH = 8;
     private final StorageDao storageDao = new StorageDaoImpl();
-
+    
     @Override
-    public User register(User user) {
+    public User register(User user) throws ValidationException {
+        if (user.getId() == null
+                || user.getLogin() == null
+                || user.getPassword() == null
+                || user.getAge() == null) {
+            throw new ValidationException("Any user value cannot be null");
+        }
+        
+        if (user.getId().toString().length() >= MIN_ID_LENGTH
+                && storageDao.get(user.getLogin()) == null
+                && user.getLogin().length() >= MIN_LOGIN_AND_PASSWORD_LENGTH
+                && user.getPassword().length() >= MIN_LOGIN_AND_PASSWORD_LENGTH
+                && user.getAge() >= MIN_USER_AGE) {
```

Comment:
Instead of having all conditions in one if statement, consider breaking it into separate if statements with specific ValidationException messages for each condition. This will help users understand what went wrong during registration.